packages/core/src/signer/types.ts

```diff
@@ -1,12 +1,23 @@
 import type { Address } from "abitype";
-import type { Hash, Hex } from "viem";
+import type { Hex } from "viem";
 import type { SignTypedDataParams } from "../account/types.js";
 
-export interface SmartAccountSigner {
+export interface AuthSmartAccountSigner<Inner, AuthParams, AuthDetails>
```

Comment:
nit: maybe we call this `SmartAccountAuthenticator`?

Reply:
kinds makes me consider calling this package `aa-auth`.